addons/website_mass_mailing/static/src/js/website_mass_mailing.js

```diff
@@ -86,6 +96,32 @@ publicWidget.registry.subscribe = publicWidget.Widget.extend({
         return this.$target.closest('[data-snippet=s_newsletter_block').data('list-id') || this.$target.data('list-id');
     },
 
+    /**
+     * Build an error message dynamically and add it to the snippet
+     * (snippets are static after creation, t-if is not an option)
+     *
+     * @returns {HTMLElement} the warning div
+     */
+    _displayListDoesNotExist() {
+        // Build from standard elements instead of inserting HTML to prevent XSS from translations
+        // Result should be:
+        //<div class="alert alert-warning text-center">
+        //  <p class="m-0">This list was removed from the database. Edit the target list or delete this block.</p>
+        //</div>
+        const insertedDiv = document.createElement('div');
+        const insertedParagraph = document.createElement('p');
+        const insertedWarningSign = document.createElement('span');
+        const insertedText = document.createTextNode(_t('This block is not linked to a Newsletter. Pick one or remove this block.'));
+        insertedDiv.classList.add('alert', 'alert-warning', 'text-center', 'mb-2');
+        insertedParagraph.classList.add('m-0');
+        insertedWarningSign.classList.add('fa', 'fa-exclamation-triangle', 'mx-1');
+        insertedParagraph.appendChild(insertedWarningSign);
+        insertedParagraph.appendChild(insertedText);
+        insertedDiv.appendChild(insertedParagraph);
```

Comment:
Note that it can be simplified
```js
const insertedDiv = new DOMParser().parseFromString(
    `<div class="alert alert-warning text-center">
        <p class="m-0"></p>
    </div>`, 'text/xml');
insertedDiv.querySelector('p').innerText = _t('This list was removed from the database. Edit the target list or delete this block.');
```

Reply:
Hm, was mildly concerned about using straight-up parsing but I guess there's no risk actually..

Thanks for the comments :)

Reply:
as long as user can't control what we give to `parseFromString`, np :)